### DISCUSSION TITLE
v1.110.0 - 40_000 stars release

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#11374] [Announcements]
# v1.110.0 - 40_000 stars release

Hello, Alex here. I want to thank you all again for your feedback and support. I found a design I made in April 2022 to show off the achievement to my family of getting 1000 stars on GitHub when Immich was still a baby. Now, the project has grown to join the big boy club. Thank you for all the stars and love that you sent us. We are forever grateful.

<p align="center">
<img width="500" src="https://md.immich.cloud/uploads/0272ab9f-92c1-4a60-b9d2-d5b80d2e8c3e.png"/>
</p>

I hope this release helps clarify our intentions and mission surrounding the financial support for Immich and open-source software in general. We could not have come back with a better solution without you. 

If you are interested in watching, here is a video from FUTO addressing many topics and concerns that have been discussed over the past week. I hope you like it and find it informative. [Youtube](https://www.youtube.com/watch?v=YdwG6SHeZEA)

> [!WARNING] 
> If you use CUDA for machine learning, the minimum NVIDIA driver version is now 545 from 535.

## Highlights

Welcome to release `v1.110.0` of Immich. This release addresses the wording change in our introduction of a way to purchase Immich, as well as many bug fixes and enhancements. Some of the highlights include

- [Migration to Immich hosted map tile server](#New-Immich-Hosted-Maps)
- [Wording changes to purchase mechanism.](#Wording-and-minor-changes-to-purchase-mechanism)
- [Claiming product keys mechanism for past financial donors of the project.](#Claiming-a-product-key)
- [TLS client certificate for the mobile app.](#TLS-client-certificate-for-the-mobile-app)

### New Immich-Hosted Maps
Previously, our maps were hosted by a third-party provider; however, going forward, that will no longer be the case. We are now hosting the map tiles ourselves at tiles.immich.cloud, and therefore your web or mobile client will no longer reach out to other third parties when browsing the map! This change should be pretty transparent, or if anything hopefully feel faster, however if you do experience problems with the map after this release, please reach out to us on Discord or GitHub!

<details>
<summary>Map light mode (expand to preview)</summary>

![map-light-mode](https://md.immich.cloud/uploads/91aac165-1e9a-4c91-a467-7a0a005f31d6.png)
</details>
<details>
<summary>Map dark mode (expand to preview)</summary>

![map-dark-mode](https://md.immich.cloud/uploads/28cf59f7-ef84-4c96-b7b0-e73d46812ce2.png)
</details>

### Wording and minor changes to the purchase mechanism.

As mentioned in our [follow-up announcement](https://github.com/immich-app/immich/discussions/11313) about the wording changes for the purchase mechanism, we made changes to the following items:

- The `licensed` or `unlicensed` word will now be called _"The-Words-Which-Must-Not-Be-Named"_ in Immich's world 😂
- Replaced `"The-Words-Which-Must-Not-Be-Named"` indication with a Buy button.
- The buy button will only show when the account age is older than 14 days.
- Added options to allow users to hide the Buy button for 30 days or indefinitely. If you hide the button indefinitely and later change your mind, you can find the option to buy Immich directly under `User Settings > Purchase Settings`
- Added more information to clarify our mission and intentions.
- Replaced the `"The-Words-Which-Must-Not-Be-Named"` badge with a supporter badge.
- Added option to hide the supporter badge.

#### Hide buy button options
<img width="500"  src="https://github.com/user-attachments/assets/15bb1936-8691-429b-84e7-3e123cdb3165" alt="buy button options screenshot"/>

#### Supporter badge

Give it a hover for extra shininess!

![supporter-badge](https://github.com/user-attachments/assets/81d8215a-3783-4362-96aa-c2bac49bee53)

##### After activation
<img width="600" alt="image" src="https://github.com/user-attachments/assets/bc02f4d2-c86d-496c-8a0c-b45e5809c39e">

##### User settings 
<img width="600" alt="image" src="https://github.com/user-attachments/assets/2a8fa91f-6f69-4fb8-91da-b32246859688">

### Claiming a product key

For the past financial donors of the project, you can claim your product keys from [https://buy.immich.app/claim](https://buy.immich.app/claim) using your GitHub credentials.

<img width="500" alt="image" src="https://github.com/user-attachments/assets/0df8a937-71a2-4db8-a65b-2c222f9b35dc">

Thank you for your support.

### TLS client certificate for the mobile app

With @yjiang-c's contribution, you can now add a custom TLS certificate on the mobile app if your networking is configured to accept one. 

https://github.com/immich-app/immich/assets/66518086/c36d548f-a473-4f0c-93ab-23e51cce6f58

### Better OpenVINO support

OpenVINO is now faster and works for a wider range of hardware. Previously, it was common for it to fail for either face detection or smart search depending on hardware. Since this involved making changes in the models, if you're using OpenVINO, please delete your model cache volume so it downloads the updated models. You only need to do this if you're using OpenVINO.

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

### 🗄️ Server
* fix(server): static mail attachment extension by @danieldietzler in https://github.com/immich-app/immich/pull/11254
* feat(server): accepted video containers by @mertalev in https://github.com/immich-app/immich/pull/11274
* fix(server): exif description whitespace handling by @michelheusschen in https://github.com/immich-app/immich/pull/11249
* fix(server): correct openapi response type for getServerLicense() by @michelheusschen in https://github.com/immich-app/immich/pull/11261
* fix(server): DateTimeOriginal overwrite issue with sidecar file by @michelheusschen in https://github.com/immich-app/immich/pull/11306
* fix(deps): exiftool-vendored by @jrasm91 in https://github.com/immich-app/immich/pull/11338
* feat: people infinite scroll by @michelheusschen in https://github.com/immich-app/immich/pull/11326
* feat(server): Allow activating non-admin user with server license by @stephen304 in https://github.com/immich-app/immich/pull/11206
### 📱 Mobile
* feat(mobile): remove asset from album in gallery view by @fyfrey in https://github.com/immich-app/immich/pull/11184
* fix(mobile): negative coordinate input by @michelheusschen in https://github.com/immich-app/immich/pull/11292
* fix(mobile): native share functionality on iPad by @Saschl in https://github.com/immich-app/immich/pull/11294
* feat(mobile): Adding setting in mobile app to TLS client certificate by @yjiang-c in https://github.com/immich-app/immich/pull/10860
### 🖥️ Web
* fix(web): use fixed position for download and upload panel by @michelheusschen in https://github.com/immich-app/immich/pull/11279
* feat(web): coordinate input for asset location by @michelheusschen in https://github.com/immich-app/immich/pull/11291
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/11165
### 🧠 Machine Learning
* feat(ml): improved ARM-NN support by @fyfrey in https://github.com/immich-app/immich/pull/11233
* fix(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/10740
### 📓 Documentation
* docs: 40k stars! ⭐  by @danieldietzler in https://github.com/immich-app/immich/pull/11265
* docs: corrected container port for immich_microservices by @itsnikhil in https://github.com/immich-app/immich/pull/11170
* chore(docs): compress homepage screenshots, 10x smaller by @zackpollard in https://github.com/immich-app/immich/pull/11347
* docs: add product key to roadmap by @aviv926 in https://github.com/immich-app/immich/pull/11351
* fix: "acess" should be "access" by @imakida in https://github.com/immich-app/immich/pull/11363
* docs: Add missing info to asset types and storage locations by @aviv926 in https://github.com/immich-app/immich/pull/11358
### 🔨 Maintenance
* chore(deps): bump docker/setup-buildx-action from 3.4.0 to 3.5.0 by @dependabot in https://github.com/immich-app/immich/pull/11284
* chore(deps): bump docker/setup-qemu-action from 3.1.0 to 3.2.0 by @dependabot in https://github.com/immich-app/immich/pull/11283
* chore(deps): bump docker/build-push-action from 6.3.0 to 6.5.0 by @dependabot in https://github.com/immich-app/immich/pull/11282
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/11299
* chore(deps): update base-image to v20240723 (major) by @renovate in https://github.com/immich-app/immich/pull/11311
* chore(deps): update node by @renovate in https://github.com/immich-app/immich/pull/11300
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/11310
* chore(deps): update node by @renovate in https://github.com/immich-app/immich/pull/11322
* chore(deps): update redis:6.2-alpine docker digest to e3b17ba by @renovate in https://github.com/immich-app/immich/pull/11303
### Other changes
* chore(docs): change some wording by @alextran1502 in https://github.com/immich-app/immich/pull/11201
* chore(mobile): post release task by @alextran1502 in https://github.com/immich-app/immich/pull/11220
* fix(mobile): changed "x jaren" to "x jaar" in dutch app translations by @JordyEGNL in https://github.com/immich-app/immich/pull/11371

## New Contributors
* @itsnikhil made their first contribution in https://github.com/immich-app/immich/pull/11170
* @imakida made their first contribution in https://github.com/immich-app/immich/pull/11363
* @yjiang-c made their first contribution in https://github.com/immich-app/immich/pull/10860

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.109.2...v1.110.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.110.0'>v1.110.0 - 40_000 stars release</a>.</em>

**Comment by Morethanevil** [comment]
Is it possible to host the map tiles ourselves? Why does the map need an external server?

A connection to static.immich.cloud was made too, why that? 🤔

**Reply by zackpollard** [reply]
The tiles require a 120GB file, so for most people that is not feasible. However it is possible to modify the maps styles.json and host your own tileserver if you really wanted to. The connection to static.immich.app is to retrieve some other map assets, specifically fonts and sprites for the map rendering.

**Reply by Morethanevil** [reply]
The size of the tiles would not be a problem, but I don't like custom modifications :)

Thank you for the clarification

**Reply by tannermeade** [reply]
> The tiles require a 120GB file, so for most people that is not feasible. However it is possible to modify the maps styles.json and host your own tileserver if you really wanted to. The connection to static.immich.app is to retrieve some other map assets, specifically fonts and sprites for the map rendering.

Is the tile server you use this server? https://github.com/immich-app/geoshenanigans

If so would love to have more info about it.

**Reply by zackpollard** [reply]
> > The tiles require a 120GB file, so for most people that is not feasible. However it is possible to modify the maps styles.json and host your own tileserver if you really wanted to. The connection to static.immich.app is to retrieve some other map assets, specifically fonts and sprites for the map rendering.
> 
> Is the tile server you use this server? https://github.com/immich-app/geoshenanigans
> 
> If so would love to have more info about it.

That is specifically for setting it up in cloudflare workers, we will post some instructions next week in the docs about hosting your own tileserver

**Reply by Sjustein** [reply]
> > > The tiles require a 120GB file, so for most people that is not feasible. However it is possible to modify the maps styles.json and host your own tileserver if you really wanted to. The connection to static.immich.app is to retrieve some other map assets, specifically fonts and sprites for the map rendering.
> > 
> > 
> > Is the tile server you use this server? https://github.com/immich-app/geoshenanigans
> > If so would love to have more info about it.
> 
> That is specifically for setting it up in cloudflare workers, we will post some instructions next week in the docs about hosting your own tileserver

Might have missed it, but were these instructions ever posted?

**Reply by zackpollard** [reply]
> > > > The tiles require a 120GB file, so for most people that is not feasible. However it is possible to modify the maps styles.json and host your own tileserver if you really wanted to. The connection to static.immich.app is to retrieve some other map assets, specifically fonts and sprites for the map rendering.
> > > 
> > > 
> > > Is the tile server you use this server? https://github.com/immich-app/geoshenanigans
> > > If so would love to have more info about it.
> > 
> > That is specifically for setting it up in cloudflare workers, we will post some instructions next week in the docs about hosting your own tileserver
> 
> Might have missed it, but were these instructions ever posted?

No, we changed approach and wanted to bundle it directly into Immich as an option so it's as simple as just changing a setting to get it working fully locally (and downloading a 120GB file). This work hasn't been completed yet.

**Comment by suderman** [comment]
Purchased! Thanks, team.

**Comment by wlmeng11** [comment]
Thank you for the thoughtful update, just purchased a product key! Looking forward to the continued success of the project :)

**Comment by titro** [comment]
yeeees nice work. OpenVINO now finally running without errors :-)

**Comment by scoobybejesus** [comment]
After deleting the model cache volume, should any jobs be re-run?

**Reply by alextran1502** [reply]
No, model cache volume store the machine learning models, it should automatically redownload the next time there is a machine learning job get queued

**Comment by jhbruhn** [comment]
I was a supporter on liberapay before, is there also a way to claim a license? No worries if not and thanks for these changes, I think all of this was handled exceptionally well!

**Reply by jrasm91** [reply]
We can probably get you one if you send us an email to `claim@immich.app` with some details, proof of payment etc.

**Comment by eenturk** [comment]
I am a early donor and i also bought the license. When the stable version releases i will go for the server license my fam is sick of the cloud :)

Good job broski's glad this project got a sick turnaround (in a good way)

**Comment by vitormarcal** [comment]
It would be interesting if the calls made to static.immich.cloud and tiles.immich.cloud were made by the backend as a proxy. So as not to send the origin and referer header with the application URL instance.

**Comment by entonjackson** [comment]
Still no reply from FUTO in regards of receiving an invoice for the purchase...

**Reply by alextran1502** [reply]
Sorry about that, can you send me your email you used for purchase. i can send you the invoice

**Comment by jaykayenn** [comment]
What's with the requirement for NVIDIA 545? Isn't 535 still the most widely used stable version in most distros?

**Reply by mertalev** [reply]
> What's with the requirement for NVIDIA 545? Isn't 535 still the most widely used stable version in most distros? 

The dependency we use updated to require cuDNN 9, so we updated the base image to the most compatible CUDA image that ships with this. I don't foresee needing to increase the minimum again any time soon.

**Reply by jaykayenn** [reply]
This seems to suggest otherwise: https://docs.nvidia.com/deeplearning/cudnn/latest/reference/support-matrix.html
Installing an experimental/unsupported driver branch to run Immich just seems like a big ask to me. Perhaps revise the release to require cudnn9 specifically, rather than asking users to change their GPU driver?

**Reply by mertalev** [reply]
> This seems to suggest otherwise: [docs.nvidia.com/deeplearning/cudnn/latest/reference/support-matrix.html](https://docs.nvidia.com/deeplearning/cudnn/latest/reference/support-matrix.html)

This isn't related to what I said, though. If you look at [NVIDIA's official CUDA images](https://hub.docker.com/r/nvidia/cuda), the image with the lowest CUDA version that ships with cuDNN 9 is 12.3.2, which in turn required 545 or higher. If there were another official image with cuDNN 9 that uses CUDA 12.2, I'd be happy to use that instead.

**Reply by j-seixas** [reply]
Yeah but this makes it impossible for those who are running Debian or debian based home servers like OMV. Currently the version available is 535, even on the "sid" unstable channel..
Guess I will stick to v1.109 for now

**Reply by j-seixas** [reply]
Going back to this, according to the [requirements page](https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements) of the ONNX runtime it should work with any 12.x cuda version, which technically be `>=525.60.13 || >=527.41` nvidia driver depending on platform 🤔 

And it should be compatible with the [minor version compability](https://docs.nvidia.com/deploy/cuda-compatibility/#minor-version-compatibility)

**Reply by mertalev** [reply]
Hmm, I wasn't aware that some distros lag behind this much. NVIDIA already considers 550 their production branch for Linux...

I also haven't tested that it actually requires 545. From the link you shared, 535 might actually work fine due to CUDA minor version compatibility. Can anyone running 535 confirm?

**Reply by j-seixas** [reply]
@mertalev I will try when I have time. Rollbacking this version if required should be only setting the version on the docker image right?

**Reply by mertalev** [reply]
Yes, that's right.

**Reply by j-seixas** [reply]
Tried it but it seems to **not** work

**Reply by bunkerman** [reply]
Hi, I'm using OMV/Debian 12 with NVIDIA v535. What error do you get? I think mine is working, I just get an out of memory error, because of my 2GB Quadro card.

**Reply by bunkerman** [reply]
No @j-seixas you're right, it doesn't work with 535 I get a runtime exception. I rolled back to v1.109.2
Any way to add a NVIDIA repo to Debian? I don't want to run the run installer.

**Reply by jaykayenn** [reply]
535 is still the current supported stable driver on Ubuntu and derivatives. 545 is unstable branch, and 550 is only in testing PPA. Immich 1.110 with 535 breaks all CUDA features.

**Reply by j-seixas** [reply]
Apparently the required version was lowered on the latest release v1.115 to 535 -> https://github.com/immich-app/immich/commit/ad58d7e23edd993a8920c48e50932508b7be1d59 but was undocummented in the release notes. As someone tried it yet?

**Reply by saydo17** [reply]
> Apparently the required version was lowered on the latest release v1.115 to 535 -> [ad58d7e](https://github.com/immich-app/immich/commit/ad58d7e23edd993a8920c48e50932508b7be1d59) but was undocummented in the release notes. As someone tried it yet?

Thank you so much for that!

**Comment by IDDQD69** [comment]
I like these changes. Will purchase for support.

**Comment by diyoyo** [comment]
Last time, I said I didn't care about the wording, but this time, I realize I'm confused, it doesn't make sense anymore. What are we buying? If you buy something, there has to be a difference between before and after. What is wrong with "Sponsoring"?
I used to sponsor directly through github before the FUTO era, and (sorry, watching the buyout video replay is still on my todo list) I actually thought that FUTO was now "kind of" the bigger sponsor.

I understand the need for a business model, but clearly there is none yet, so don't anticipate with the wording. If you want to not make it free anymore, "fine" (i mean, I'll be sad and it defeats the purpose but if that's the viable decision, so be it) then put a pricetag before hand. 

When there will be something to buy, call it "buy", but in the meantime, what am I buying? Or did I miss some "premium" features? It's not even funny anymore when you have to choose between $100 and $25, and you know these tags are meaningless. I hate meaningless things, if not in the context of a game. It looks like "fake" to me, now. 
I want to pay for something that makes sense, like sponsoring you on your journey to bring this to the rest of the world, not buying fake/empty/wind products.
Please let me (us?) be just sponsors.
(Oh And please fix this scrollbar that makes me "hide forever" the button)

**Reply by EstrianTech** [reply]
Agree, "Support Immich" seems like more correct wording, especially since you are not receiving anything for your purchase. In that case the different supporter levels make even less sense, why not just allow people to choose how much they want to contribute over a certain amount?

**Comment by diyoyo** [comment]
Other than that. I don't know whether it comes from this update or not, but Huge congrats and BIG Thank you for the fix in the way people are being displayed: the progressive loading of all people is perfect!!!!🥳🥳👍
I don't have to manually edit the JS file to change the limit in the container :D, then commit a modified version etc. so it's very cool.

Just another reason for me to celebrate and sponsor it, when there will be a sponsor button.

**Comment by MadMan247** [comment]
Hello.
I notice that despite the statements in this release, the official immich website's blogs still use, as you put it, the "The-Words-Which-Must-Not-Be-Named". Let alone, it is the latest blog, and thus the first that shows up on the blogs section. I believe that it would be wise to update this.
https://immich.app/blog/2024/immich-licensing
please remedy.
